pkg/reconciler/knativeeventing/common/images.go

```diff
@@ -92,16 +93,19 @@ func updateDeploymentImage(deployment *appsv1.Deployment, registry *eventingv1al
 	containers := deployment.Spec.Template.Spec.Containers
 	for index := range containers {
 		container := &containers[index]
-		newImage := getNewImage(registry, container.Name)
+		newImage := getNewImage(registry, container.Name, deployment.Name)
 		if newImage != "" {
 			updateContainer(container, newImage, log)
 		}
 	}
 	log.Debugw("Finished updating images", "name", deployment.GetName(), "containers", deployment.Spec.Template.Spec.Containers)
 }
 
-func getNewImage(registry *eventingv1alpha1.Registry, containerName string) string {
-	overrideImage := registry.Override[containerName]
+func getNewImage(registry *eventingv1alpha1.Registry, containerName, deploymentName string) string {
+	overrideImage := registry.Override[deploymentName+delimiter+containerName]
```

Comment:
Re-using the same override map for deployment and deployment-container style overrides could cause issues if the deployment name contains -, which seems to be possible.

Thoughts on creating two maps, one for container-only overrides and one for deployment+container?

Reply:
I have changed the delimiter to `/`.